### PR TITLE
Error handling

### PR DESCRIPTION
**Pull request by GhostInAMachine**

### CODE REVIEW

File: src/HazelcastClient.ts [hunk 1]
```diff
@@ -44,8 +44,8 @@ class HazelcastClient {
             return this.partitionService.initialize();
         }).then(() => {
             deferred.resolve(this);
-        }).catch(() => {
-            deferred.reject('Client failed to start');
+        }).catch((e) => {
+            deferred.reject(e);
```
Comment (mustafaiman) [inline src/HazelcastClient.ts]: Propagating the error to upper level is already default behaviour. This `.catch` is unnecessary. Correct me if I am wrong.

Comment (GhostInAMachine) [inline src/HazelcastClient.ts]: Before the change in case of the error you would only get a failure message without any explanation.

Comment (mustafaiman) [inline src/HazelcastClient.ts]: I see now. Ok.

**Comment by mustafaiman** [comment]
:+1:

**Comment by asimarslan** [comment]
:+1:

**Pull request merged**